beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3127,6 +3173,19 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                 let delay =
                     get_slot_delay_ms(timestamp_now(), block.block.slot(), &self.slot_clock);
                 metrics::observe_duration(&metrics::BLOCK_AVAILABILITY_DELAY, delay);
+                if let Some(slasher) = self.slasher.as_ref() {
+                    if let Some(blobs) = block.block.blobs() {
+                        let block_header = block.block.block().signed_block_header();
+                        for blob_sidecar in blobs {
+                            let blob_header = blob_sidecar.signed_block_header.clone();
+                            // block has already been added to slasher, so only add if header in blob
+                            // sidecar is different to block header
+                            if blob_header != block_header {
```

Comment:
I'm not familiar with slash code, but under normal circumstances we'd be cloning and adding the same `SignedBlockHeader` 0  to 6 times. Is it worth filtering this to save on memory and processing time?

Reply:
I think cloning the `SignedBlockHeader` is okay since its quite small (~200 bytes), my thinking was that the slasher is already heavy, might as well not give it dupes

Reply:
We were missing adding some slashable headers that I realised because of this comment. Fixed in [1fa71ec](https://github.com/sigp/lighthouse/pull/4900/commits/1fa71ece2b55b9dc5b50e8b1b1222b3261680763)
Thanks @jimmygchen